### PR TITLE
TEM-171: flags

### PR DESCRIPTION
**Pull request by bobheadxi**
## :construction_worker: Purpose

replace env config with flags

## :rocket: Changes

The new `temporal`:

```
Temporal is an easy-to-use interface into distributed and decentralized storage technologies for personal and enterprise use cases.

USAGE:

  temporal [COMMAND]

COMMANDS:

  api      start Temporal api server
  help     show help text
  init     initialize blank Temporal configuration
  krab     runs the krab service
  migrate  run database migrations
  queue    execute commands for various queues
  version  show Temporal version

OPTIONS:

  -api.port string
        set port to expose API on (default "6767")
  -config string
        path to Temporal configuration
  -db.migrate
        toggle whether a database migration should occur
  -db.no_ssl
        toggle SSL connection with database
  -dev
        toggle dev mode
  -grpc.no_ssl
        toggle SSL connection with GRPC services
```

also deprecated a few commands (notably `*-insecure` stuff)

## :warning: Breaking Changes

many env things must be set via flags now - see documentation

### CODE REVIEW

**Comment by postables** [comment]
Getting an issue running the bucket creation command
```
Use 'temporal help [command]' to see CLI documentation.solidity@dark:~/go/src/github.com/RTradeLtd/Temporal$ ./temporal --config testenv/config.json bucket new filesuploadbucket
[filesuploadbucket]
invalid invocation for 'temporal bucket new filesuploadbucket'

Use 'temporal help [command]' to see CLI documentation.s
```

There's also a panic error when closing the API
```
Starting API service with TLS...
press CTRL+C to stop processing and close queue resources



^Cpanic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x8141c2]

goroutine 1 [running]:
github.com/RTradeLtd/Temporal/vendor/google.golang.org/grpc.(*ClientConn).Close(0x0, 0x0, 0x0)
	/home/solidity/go/src/github.com/RTradeLtd/Temporal/vendor/google.golang.org/grpc/clientconn.go:822 +0x52
github.com/RTradeLtd/Temporal/grpc-clients.(*SignerClient).Close(0xc000214620)
	/home/solidity/go/src/github.com/RTradeLtd/Temporal/grpc-clients/eth.go:38 +0x2f
github.com/RTradeLtd/Temporal/grpc-clients.(*SignerClient).Close-fm()
	/home/solidity/go/src/github.com/RTradeLtd/Temporal/cmd/temporal/main.go:141 +0x2a
main.glob..func1.1(0xc000214640, 0x3, 0x4)
	/home/solidity/go/src/github.com/RTradeLtd/Temporal/cmd/temporal/main.go:170 +0x33
main.glob..func1(0xc0003620e0, 0x18, 0xc000362100, 0x17, 0xc0004d8920, 0x9, 0x0, 0x0, 0x0, 0x0, ...)
	/home/solidity/go/src/github.com/RTradeLtd/Temporal/cmd/temporal/main.go:201 +0x688
github.com/RTradeLtd/Temporal/vendor/github.com/RTradeLtd/cmd.run(0xc000364420, 0xc0003620e0, 0x18, 0xc000362100, 0x17, 0xc0004d8920, 0x9, 0x0, 0x0, 0x0, ...)
	/home/solidity/go/src/github.com/RTradeLtd/Temporal/vendor/github.com/RTradeLtd/cmd/utils.go:45 +0x2b6
github.com/RTradeLtd/Temporal/vendor/github.com/RTradeLtd/cmd.(*App).Run(0xc000378000, 0xc0003620e0, 0x18, 0xc000362100, 0x17, 0xc0004d8920, 0x9, 0x0, 0x0, 0x0, ...)
	/home/solidity/go/src/github.com/RTradeLtd/Temporal/vendor/github.com/RTradeLtd/cmd/app.go:108 +0xe2
main.main()
	/home/solidity/go/src/github.com/RTradeLtd/Temporal/cmd/temporal/main.go:674 +0x61a
```

Excellent changes though :D

**Comment by postables** [comment]
Actually after re-reading the error message given when closing the API, it doesn't look like it's that much of an issue, since it's trying to close a non-existing gRPC connection.

**Comment by postables** [comment]
I think the tests are failing due to the `gorm` library not using versioning, and a bad update to their master branch breaking the code-base. I've forked the library and created a release at the code we were using previously to prevent this from happening again.

See https://github.com/RTradeLtd/gorm/releases/tag/v2.0.0rc1

**Comment by bobheadxi** [comment]
> since it's trying to close a non-existing gRPC connection.

hm strange

File: rtfscluster/rtfs_cluster.go [hunk 1]
```diff
@@ -55,8 +55,9 @@ func (cm *ClusterManager) GenClient() error {
 func (cm *ClusterManager) ParseLocalStatusAllAndSync() ([]gocid.Cid, error) {
 	// this will hold all the cids that have been synced
 	var syncedCids []gocid.Cid
-	// only fetch the local status for all pins
-	pinInfo, err := cm.Client.StatusAll(true)
+	// only fetch the local status for all pins - todo: double check if this is
+	// the right TrackerStatusError to user
+	pinInfo, err := cm.Client.StatusAll(api.TrackerStatusError, true)
```
Comment (bobheadxi) [inline rtfscluster/rtfs_cluster.go]: @postables weird dependency updates from vendor rebuild, this is one - `api.TrackerStatusError` is just a guess, not sure if right?

File: rtfscluster/rtfs_cluster.go [hunk 2]
```diff
@@ -105,7 +106,8 @@ func (cm *ClusterManager) RemovePinFromCluster(cidString string) error {
 // FetchLocalStatus is used to fetch the local status of all pins
 func (cm *ClusterManager) FetchLocalStatus() (map[gocid.Cid]string, error) {
 	var response = make(map[gocid.Cid]string)
-	pinInfo, err := cm.Client.StatusAll(true)
+	// todo: doublecheck if this is the right api.TrackerStatus to use
+	pinInfo, err := cm.Client.StatusAll(api.TrackerStatusPinned, true)
```
Comment (bobheadxi) [inline rtfscluster/rtfs_cluster.go]: same here (similar to https://github.com/RTradeLtd/Temporal/pull/287/files#r246644893)

File: rtfscluster/rtfs_cluster.go (hunk 1 above)
Comment (bobheadxi) [reply rtfscluster/rtfs_cluster.go]: Ill leave it for now, we can remove in another PR

Comment (postables) [reply rtfscluster/rtfs_cluster.go]: Sounds good

**Pull request merged**